### ISSUE TITLE
relative paths in the build.js should be relative to the process/shell cwd not the build.js file itself.

### ISSUE THREAD

**Issue by sterpe** [jrburke/requirejs#1423]
```
In general, if it is a path, it is relative to the build.js file used to hold the build options, or if just using command line arguments, relative to the current working directory.
```

This is so strange and counterintuitive, and not at all UNIXy.

**Comment by jrburke** [comment]
How so? If a path is in the build.js, it is relative to that file. That is good because it means it does not matter where the optimization call is done, if fed the build.js file, it will work out. 

If on the command line, it is relative to current working directory. ls works similarly, asking for a relative path to ls uses cwd.

**Comment by sterpe** [comment]
It's just not the way an rc type options file would typically work.  If I'm on the commandline with something like this:

r.js -o foo/bar/quux.js 

I would still expect that any relative path defined in quux.js is relative to the directory I am currently running the executable in.  I should be able to put quux.js anywhere on a system at a whim because the options it defines feel like they should apply to the running r.js process which is happening in a specific place.

As with any config/rc file all we're saying is load the config file *from* here, which could be from anywhere, the file itself shouldn't have to be 'aware' of its position on the file system relative to the project, imo.

**Comment by sterpe** [comment]
tldr; by running the process from a specific directory the user is already telling r.js where they would like to root the process, relative paths should be calculated accordingly.
